### ISSUE TITLE
Do not detect type mismatch with string concatenation

### ISSUE THREAD

**Issue by microcoder** [Microsoft/pyright#70]
**Describe the bug**
Pyright does not detect type mismatch with string concatenation

**To Reproduce**

```python
def test(fname: str, salary: int) -> str:
    return fname.title() + ': ' + salary


test('vasya', 100)
```

**Expected behavior**
variable *salary* should be highlighted

**Comment by erictraut** [comment]
Thanks for the bug report. There is TODO comment in the code related to this issue. Pyright currently validates that objects have an __add__ operator, but it doesn't yet enforce the type constraints for the operator override, which is why it doesn't report an error in this particular case.

**Comment by microcoder** [comment]
@erictraut, Hi! Many thanks!

**Comment by erictraut** [comment]
This is now implemented in version 1.0.15, which I just published.
